### ISSUE TITLE
Missing the species names and reaction family when RMG's unable to apply an atom action

### ISSUE THREAD

**Issue by alongd** [ReactionMechanismGenerator/RMG-Py#811]
It would be helpful to debug when a reaction family recipe action fails, if the species and the family were known. I got the following trace when RMG tried to decrement a lone pair from a 5 valance nitrogen (with no lone pairs):
```
Traceback (most recent call last):
  File "D:\Code\RMG-Py\rmgpy\scoop_framework\util.py", line 112, in __call__
    return self.myfn(*args, **kwargs)
  File "D:\Code\RMG-Py\rmgpy\rmg\react.py", line 96, in reactMolecules
    rxns = family.generateReactions(molecules)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1349, in generateReactions
    reactionList.extend(self.__generateReactions(reactants, forward=False))
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1416, in __generateReactions
    productStructures = self.__generateProductStructures(reactantStructures, [map], forward)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1200, in __generateProductStructures
    productStructures = self.applyRecipe(reactantStructures, forward=forward)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1091, in applyRecipe
    self.reverseRecipe.applyForward(reactantStructure, unique)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 292, in applyForward
    return self.__apply(struct, True, unique)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 282, in __apply
    atom.applyAction(['LOSE_PAIR', label, 1])
  File "rmgpy\molecule\molecule.py", line 389, in rmgpy.molecule.molecule.Atom.applyAction (build\Release\pyrex\rmgpy\molecule\molecule.c:8092)
    for i in range(abs(action[2])): self.decrementLonePairs()
  File "rmgpy\molecule\molecule.py", line 356, in rmgpy.molecule.molecule.Atom.decrementLonePairs (build\Release\pyrex\rmgpy\molecule\molecule.c:7358)
    raise gr.ActionError('Unable to update Atom due to LOSE_PAIR action: Invalid lone electron pairs set "{0}".'.format(self.lonePairs))
ActionError: Unable to update Atom due to LOSE_PAIR action: Invalid lone electron pairs set "-1".

Traceback (most recent call last):
  File "D:\Code\RMG-Py\/rmg.py", line 152, in <module>
    rmg.execute(**kwargs)
  File "D:\Code\RMG-Py\rmgpy\rmg\main.py", line 533, in execute
    bimolecularReact=self.bimolecularReact)
  File "D:\Code\RMG-Py\rmgpy\rmg\model.py", line 621, in enlarge
    rxns = reactAll(self.core.species, numOldCoreSpecies, unimolecularReact, bimolecularReact)
  File "D:\Code\RMG-Py\rmgpy\rmg\react.py", line 155, in reactAll
    rxns = list(react(*spcTuples))
  File "D:\Code\RMG-Py\rmgpy\rmg\react.py", line 75, in react
    combos
  File "D:\Code\RMG-Py\rmgpy\scoop_framework\util.py", line 154, in map_
    return map(WorkerWrapper(args[0]), *args[1:], **kwargs)
  File "D:\Anaconda\envs\rmg_env\lib\site-packages\scoop-0.7.2.0-py2.7.egg\scoop\fallbacks.py", line 49, in wrapper
  File "D:\Code\RMG-Py\rmgpy\scoop_framework\util.py", line 112, in __call__
    return self.myfn(*args, **kwargs)
  File "D:\Code\RMG-Py\rmgpy\rmg\react.py", line 96, in reactMolecules
    rxns = family.generateReactions(molecules)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1349, in generateReactions
    reactionList.extend(self.__generateReactions(reactants, forward=False))
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1416, in __generateReactions
    productStructures = self.__generateProductStructures(reactantStructures, [map], forward)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1200, in __generateProductStructures
    productStructures = self.applyRecipe(reactantStructures, forward=forward)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 1091, in applyRecipe
    self.reverseRecipe.applyForward(reactantStructure, unique)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 292, in applyForward
    return self.__apply(struct, True, unique)
  File "D:\Code\RMG-Py\rmgpy\data\kinetics\family.py", line 282, in __apply
    atom.applyAction(['LOSE_PAIR', label, 1])
  File "rmgpy\molecule\molecule.py", line 389, in rmgpy.molecule.molecule.Atom.applyAction (build\Release\pyrex\rmgpy\molecule\molecule.c:8092)
  File "rmgpy\molecule\molecule.py", line 356, in rmgpy.molecule.molecule.Atom.decrementLonePairs (build\Release\pyrex\rmgpy\molecule\molecule.c:7358)
rmgpy.molecule.group.ActionError: Unable to update Atom due to LOSE_PAIR action: Invalid lone electron pairs set "-1".
```

**Comment by alongd** [comment]
I added the species name as a first step, but to add the reaction family would require to change the arguments of all action functions such as `incrementLonePairs(self)`.
Any idea for a good way to implement this is welcome!

**Comment by alongd** [comment]
Correction: the above commit only added the atom, not species

**Comment by mliu49** [comment]
You could potentially add a try...except block in `rmgpy.data.kinetics.family.applyForward` and `applyReverse` ([see here](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/rmgpy/data/kinetics/family.py#L292)) to catch the `ActionError`, print the molecule and family, then raise the error.

**Comment by alongd** [comment]
thanks @mliu49 !
Please review the short code, if OK than I'll close the issue and PR.

**Comment by mliu49** [comment]
Maybe you should prepare the PR first? I'm not sure which commits you're planning on including for this. It might also be a good idea to put this with #810, so we don't pile up a bunch of pull requests with one change each. I'm even thinking about combining with #807 since they're all related to improving debug messages.

On an unrelated note, you don't need to commit the .idea folder in git, since it just contains settings for pycharm.

**Comment by alongd** [comment]
I agree, I'll wrap them all into one PR (and wait to see if more related issues pop up in the coming days as I examine the new nitrogen atom types)

**Comment by alongd** [comment]
closed via #813
